Mortgage-loan-calculator/src/app/admin-login/admin-login.component.ts

```diff
@@ -1,70 +0,0 @@
-import { Component } from '@angular/core';
-import { FormBuilder, FormControl, Validators } from '@angular/forms';
-import {AuthService} from "./services/auth.service";
-import {StorageService} from "./services/storage.service";
-const fb = new FormBuilder().nonNullable;
-@Component({
-  selector: 'app-admin-login',
-  templateUrl: './admin-login.component.html',
-  styleUrls: ['./admin-login.component.css'],
-})
-export class AdminLoginComponent {
-  showMyClass = true;
-  isLoggedIn = false;
-  isLoginFailed = false;
-  errorMessage = '';
-  roles: string[] = [];
-  constructor(private authService: AuthService, private storageService: StorageService) { }
-  loginForm = fb.group(
-    {
-      username: [''],
-      password: [
-        '',
-        [
-          Validators.required,
-          Validators.minLength(8),
-          Validators.pattern(
-            '^(?=.*[a-z])(?=.*\\d)[a-zA-Z\\d]{8,}$'
-          ),
-        ],
-      ],
-    },
-    { updateOn: 'change' }
-  );
-
-  get username() {
-    return this.loginForm.get('username') as unknown as FormControl<string>;
```

Comment:
Try using `controls` instead of `get` as it returns already typed FormControl - no need to assign types manually.